whole-tale/ngx-dashboard
# Git integration

## Problem
As a user of the WholeTale platform, it would be nice to be able to integrate my Tale more tightly with Git. This would enable me to more easily track my Tale in SCM tools like GitHub or BitBucket.
  
## Approach
Added inputs for specifying an associated Git repo in two different contexts:
1. during Tale creation 
2. from the Run view's dropdown menu
 
## How to Test  
Prerequisites: running https://github.com/whole-tale/girder_wholetale/pull/430 for API

1. Checkout this branch locally, rebuild the dashboard
2. Login to the WholeTale Dashboard
3. From the Tale Catalog, select Create New... > Create Tale from Git Repository
    * The `create-tale-modal` should display, with an additional input allowing the user to specify a Git repo
4. Enter a Git repo URL/Tale name, choose an environment, and click Create
    * You should be brought to the Run view for your new Tale
5. Navigate to Run > Files > Tale Workspace for this Tale
    * You should see that your git repo has been cloned into the Tale's workspace
6. Navigate to the Tale Catalog and create a new Tale (don't specify a git repo here)
    * You should be brought to the Run view for your new Tale
7. Expand the dropdown at the right side of the Run view and select "Connect to Git repository..."
    * A new modal should display with an input allowing the user to specify a Git repo
    * NOTE: This option still shows up if you have already specified a Git repo.. we have no easy way to detect this from the UI
8. Enter a Git repo URL and submit the form
9. Navigate to Run > Files > Tale Workspace for this Tale
    * You should see that your git repo has been cloned into the Tale's workspace
